### PR TITLE
[SPARK-31305][SQL][DOCS] Add a page to list all commands in SQL Reference

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add a page to list all commands in SQL Reference...

### Why are the changes needed?
so it's easier for user to find a specific command.

### Does this PR introduce any user-facing change?
before:
image

after:
image

image

Also move ```use database``` from query category to ddl category.

### How was this patch tested?
Manually build and check

### CODE REVIEW

File: docs/sql-ref-syntax.md [hunk 1]
```diff
@@ -19,4 +19,64 @@ license: |
   limitations under the License.
 ---
 
-Spark SQL is Apache Spark's module for working with structured data. The SQL Syntax section describes the SQL syntax in detail along with usage examples when applicable.
+Spark SQL is Apache Spark's module for working with structured data. The SQL Syntax section describes the SQL syntax in detail along with usage examples when applicable. This document provides a list of Data Definition and Data Manipulation Statements, as well as Data Retrieval and Auxiliary Statements.
+
+### Data Definition Statements
```
Comment (maropu) [inline docs/sql-ref-syntax.md]: How about `DDL Statements` for consistency? https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md#ddl-statements

File: docs/sql-ref-syntax.md [hunk 2]
```diff
@@ -19,4 +19,64 @@ license: |
...
+- [CREATE VIEW](sql-ref-syntax-ddl-create-view.html)
+- [DROP DATABASE](sql-ref-syntax-ddl-drop-database.html)
+- [DROP FUNCTION](sql-ref-syntax-ddl-drop-function.html)
+- [DROP TABLE](sql-ref-syntax-ddl-drop-table.html)
+- [DROP VIEW](sql-ref-syntax-ddl-drop-view.html)
+- [REPAIR TABLE](sql-ref-syntax-ddl-repair-table.html)
+- [TRUNCATE TABLE](sql-ref-syntax-ddl-truncate-table.html)
+- [USE DATABASE](sql-ref-syntax-qry-select-usedb.html)
+
+### Data Manipulation Statements
```
Comment (maropu) [inline docs/sql-ref-syntax.md]: ditto: DML Statements?

File: docs/sql-ref-syntax.md [hunk 3]
```diff
@@ -19,4 +19,64 @@ license: |
...
+### Data Retrieval Statements
+- [CLUSTER BY Clause](sql-ref-syntax-qry-select-clusterby.html)
+- [DISTRIBUTE BY Clause](sql-ref-syntax-qry-select-distribute-by.html)
+- [EXPLAIN](sql-ref-syntax-qry-explain.html)
+- [GROUP BY Clause](sql-ref-syntax-qry-select-groupby.html)
+- [HAVING Clause](sql-ref-syntax-qry-select-having.html)
+- [LIMIT Clause](sql-ref-syntax-qry-select-limit.html)
+- [ORDER BY Clause](sql-ref-syntax-qry-select-orderby.html)
+- [SORT BY Clause](sql-ref-syntax-qry-select-sortby.html)
+- [WHERE Clause](sql-ref-syntax-qry-select-where.html)
```
Comment (maropu) [inline docs/sql-ref-syntax.md]: We need these links to the sub-categoris of SELECT? Just linking to `sql-ref-syntax-qry-select.md` here is not enough?

Comment (huaxingao) [reply docs/sql-ref-syntax.md]: Thanks for your comment. I want to list all the sql statements on one page, so I prefer to have all the links here.

**Comment by huaxingao** [comment]
@srowen This is for 3.0. Thank you!

**Review by srowen — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Thanks, @huaxingao ! Merged to master/3.0. btw, have we done all the works about the SQL references? It seems the RC1 for 3.0 will be started in a few days.

**Comment by huaxingao** [comment]
Thanks! @maropu @srowen 
@gatorsmile wants a few more. I am working on functions right now, will probably submit a PR for UDF later today.

**Comment by maropu** [comment]
ok, thanks for the work! cc: @dongjoon-hyun

**Pull request closed without merging**